### PR TITLE
Updates for ZIO 2.0.0-RC5

### PR DESCRIPTION
**Pull request by gavares**
Before submitting pull request:
- [ ] Check if the project compiles by running `sbt compile`
- [ ] Verify docs compilation by running `sbt compileDocs`
- [ ] Check if tests pass by running `sbt test`
- [ ] Format code by running `sbt scalafmt`

### CODE REVIEW

**Comment by gavares** [comment]
Unclear to me if the docs should be updated for ZIO 2 or should be left on ZIO 1

File: armeria-backend/zio/src/main/scala/sttp/client3/armeria/zio/ArmeriaZioBackend.scala [hunk 1]
```diff
@@ -49,11 +49,13 @@ object ArmeriaZioBackend {
       .runtime[Any]
       .map(runtime => apply(runtime, newClient(options), closeFactory = true))
 
-  def managed(options: SttpBackendOptions = SttpBackendOptions.Default): TaskManaged[SttpBackend[Task, ZioStreams]] =
-    ZManaged.acquireReleaseWith(apply(options))(_.close().ignore)
+  def managed(
```
Comment (Denis-Savitsky) [inline armeria-backend/zio/src/main/scala/sttp/client3/armeria/zio/ArmeriaZioBackend.scala:52]: Does it have sense calling it managed? Maybe rename it to scoped?

Comment (gavares) [reply armeria-backend/zio/src/main/scala/sttp/client3/armeria/zio/ArmeriaZioBackend.scala:52]: I had the same thought. Happy to change it if we think that's a better name.

Comment (gavares) [reply armeria-backend/zio/src/main/scala/sttp/client3/armeria/zio/ArmeriaZioBackend.scala:52]: Softwaremill guys make a decision and let me know what you want to see here. Happy to make the changes or leave it as is.

**Comment by adamw** [comment]
Thanks, though closing in favor of #1407 :)

**Pull request closed without merging**